### DISCUSSION TITLE
Error while doing Parallel Training

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2198] [Q&A] [answered]
Hi,

I hope you all are doing well. I am facing an error while doing parallel training. 

Installation - From source 
Version --  DeePMD-kit v2.2.0b0
                   NCCL - 2.16.2
                  MPIRUN - 4.14.2
System -- DGX V100 Server
Command --
```bash
 mpirun -np 2 dp train input.json
```

Error --

![mpirun_err_1](https://user-images.githubusercontent.com/10074710/209313176-21a5ba5b-8fc0-4ec8-8465-e885f6c3bb67.png)

![mpirun_err_2](https://user-images.githubusercontent.com/10074710/209313399-ea8152a8-429e-42ef-84fa-536a533cc787.png)

![mpirun_err_3](https://user-images.githubusercontent.com/10074710/209313423-81421f60-ff1c-4327-bc8f-72911211e8c0.png)

Command -- 
```bash
CUDA_VISIBLE_DEVICES=4,5,6,7 horovodrun -np 4  dp train --mpi-log=workers input.json
```
Error --

![Screenshot 2022-12-23 at 3 07 02 PM](https://user-images.githubusercontent.com/10074710/209314102-2f403aa8-3cd0-4f8f-b250-216411fab516.png)

 
![Screenshot 2022-12-23 at 3 07 19 PM](https://user-images.githubusercontent.com/10074710/209314131-a1619087-6e2c-4934-b185-5204b326c7f3.png)

![Screenshot 2022-12-23 at 3 07 28 PM](https://user-images.githubusercontent.com/10074710/209314558-81ed2063-fb9c-47a2-a91e-89e52ae6e6ba.png)

![Screenshot 2022-12-23 at 3 08 06 PM](https://user-images.githubusercontent.com/10074710/209314810-382ad3bb-a872-4a88-84a6-fd99ee3b6038.png)

**Comment by AnuragKr** [comment, marked as answer]
@njzjz  Could you please help me figure out the above issue mentioned?

**Reply by njzjz** [reply]
For NCCL errors, you may still set the environment variable `NCCL_DEBUG=INFO` to debug it.

**Reply by AnuragKr** [reply]
After setting the environment I got the following output --

![nccl_1](https://user-images.githubusercontent.com/10074710/209646017-ace78ced-a601-4f35-a9f7-619e2408597a.png)
![nccl_2](https://user-images.githubusercontent.com/10074710/209646036-0a3635ee-b8f9-49ec-b0b2-0fc25786d989.png)
![nccl_3](https://user-images.githubusercontent.com/10074710/209646041-a7c325c1-6d34-4731-81a1-2ea9c7e6f24a.png)
![nccl_4](https://user-images.githubusercontent.com/10074710/209646044-3abbe49d-9da6-45e7-a5a9-11283efcdb1e.png)
![nccl_5](https://user-images.githubusercontent.com/10074710/209646047-66851c15-c477-4c3e-a065-e676dadd8c95.png)

**Reply by njzjz** [reply]
Isn't it the same as #1774?

**Reply by AnuragKr** [reply]
Yes, kind of it is the same, but I installed NCCL where that bug was already solved. Still, I am getting this error.
Last time also I was able to work around through conda but not via source. This time also I am facing a problem with source installation.
Sorry for creating another discussion.
